### ISSUE TITLE
Openshift Console Starts Printing same error message on endless loop

### ISSUE THREAD

**Issue by deanpeterson** [openshift/origin#4618]
I have been trying to get the mongodb replica set example to work for a few days.  I have three pods running and I start up the router and add the route to my headless mongodb server.

I add: 

<router ip>    mongodb.test.router.default.svc.cluster.local to my /etc/hosts file.  

That is a route I added for my mongodb headless service (no ClusterIP).  When I attempt to connect to the service using UMONGO with the host mongodb.test.router.default.svc.cluster.local:80, my openshift console forever begins to send this message (it never stops after just one request to connect - it keeps repeating over and over): 

E0910 12:54:38.017351   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:38.017371   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:38.892463   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:38.892524   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:38.892774   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:38.892820   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:39.250190   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:39.250232   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:39.250448   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:39.250465   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:41.211770   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:41.211835   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:41.212198   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:41.212243   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:42.628829   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:42.628884   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:42.629189   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:42.629210   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:43.017986   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:43.018050   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:43.018459   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:43.018509   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:43.893363   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:43.893422   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:43.893760   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:43.893813   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:44.250892   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:44.250935   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:44.251186   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:44.251203   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:46.212723   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:46.212769   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:46.213060   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:46.213080   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:47.629693   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:47.629787   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:47.630105   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:47.630124   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:48.019095   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:48.019136   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry
E0910 12:54:48.019446   19243 proxysocket.go:89] Couldn't find an endpoint for default/router:80-tcp: missing service entry
E0910 12:54:48.019493   19243 proxysocket.go:134] Failed to connect to balancer: missing service entry

<b>Why does this not work to connect to mongodb externally?</b>  And why does the openshift console start writing that message and never stop?

I tried a suggestion here: https://github.com/openshift/origin/blob/master/docs/debugging-openshift.md

I changed my resolv.conf file to point to 8.8.8.8.

Still no luck.

Do I need a different router configuration?  I followed the simple instructions here: https://docs.openshift.org/latest/admin_guide/install/deploy_router.html  but do I need a "Highly Available Router" to actually make the router work?  

Again, I am simply trying to connect externally to the mongodb replica set created by the example mongodb replica set template.  Am I missing something obvious here?

**Comment by deanpeterson** [comment]
I tried a suggestion here: https://github.com/openshift/origin/blob/master/docs/debugging-openshift.md

Still no luck.

**Comment by deanpeterson** [comment]
On the users email list Clayton suggested:

The errors above indicate your router isn't installed in the default
namespace (did you accidentally create the router in your test
namespace?).

Is this an all in one deployment, or are you running your test scripts
(curl, browser) from one of the machines like a laptop?  If so you can
just hit the mongo service directly via
mongo.test.svc.cluster.local:<serviceport>

<b>However...</b>
i run:

[root localhost openshift]# ./oc get services -n default

and see:

NAME              CLUSTER_IP       EXTERNAL_IP   PORT(S)    SELECTOR                  AGE
docker-registry   172.30.16.201    <none>        5000/TCP   docker-registry=default   8d
kubernetes        172.30.0.1       <none>        443/TCP    <none>                    8d
router            172.30.219.139   <none>        80/TCP     router=router             22h

The router is in the default namespace.

I try putting mongo.test.svc.cluster.local:27017 in UMONGO no luck.
I try mongodb.test.router.default.svc.cluster.local:80 and mongodb.test.router.default.svc.cluster.local:27017;
basically every url I can think might work.  No connection.  

I run: 
[root localhost openshift]# ./oc get services -n test

NAME      CLUSTER_IP   EXTERNAL_IP   PORT(S)     SELECTOR               AGE
mongodb   None         <none>        27017/TCP   name=mongodb-replica   22h

Does this extra information help at all?

Also, this is an all in one installation directly on RHEL using the latest binary.

**Comment by ncdc** [comment]
Issue was resolved on users mailing list (the router wasn't deploying because the "router" service account hadn't been created yet).
